test/optional-validation.test.js

```diff
@@ -0,0 +1,575 @@
+// This test written in mocha+should.js
+var should = require('./init.js');
+var async = require('async');
+var db, User, options, filter, whereCount = 0;
+var j = require('../');
+var ValidationError = j.ValidationError;
+
+var OPTIONS_TRUE = {validate: true};
+var OPTIONS_FALSE = {validate: false};
+var OLD_NAME = 'Bert'
+var NEW_NAME = 'Ernie';
+var INVALID_DATA = {name: null};
+var VALID_DATA = {name: OLD_NAME};
+
+describe('optional-validation', function () {
+
+  before(function (done) {
+    db = getSchema();
+    User = db.define('User', {
+        seq: {type: Number, index: true},
+        name: {type: String, index: true, sort: true},
+        email: {type: String, index: true},
+        birthday: {type: Date, index: true},
+        role: {type: String, index: true},
+        order: {type: Number, index: true, sort: true},
+        vip: {type: Boolean}
+      }, { forceId: true, strict: true });
+
+    filter = {fields: ['name', 'id']};
+
+    db.automigrate(['User'], done);
+
+  });
+
+  beforeEach(function (done) {
+    User.destroyAll(function () {
+      delete User.validations;
+      User.validatesPresenceOf('name');
+      done();
+    });
+  });
+
+  function expectValidationError(done) {
+    return function (err, result) {
+      should.exist(err);
+      err.should.be.instanceOf(Error);
+      err.should.be.instanceOf(ValidationError);
+      done();
+    }
+  }
+
+  function expectCreateSuccess(done, incompleteData) {
+    if (incompleteData === undefined) incompleteData = false;
+    return function (err, d) {
+      if (err) return done(err);
+      d.should.be.instanceOf(User);
+      if (!incompleteData) {
+        d.name.should.eql(OLD_NAME);
```

Comment:
I find this suspicious. The callback receives an updated model instance, so I would expect the name to equal to the new name? Unless `OLD_NAME` is referring to the value used for `create()` call, in which case a name like `INITIAL_NAME` would be probably better?